eng/helix/content/RunTests/Program.cs

```diff
@@ -0,0 +1,180 @@
+            var dotnetRoot = Environment.GetEnvironmentVariable("DOTNET_ROOT");
+
+            // Rename default.NuGet.config to NuGet.config if there is not a custom one from the project
+            if (!File.Exists("NuGet.config"))
+            {
+                File.Copy("default.NuGet.config", "NuGet.config");
+            }
+
+            var environmentVariables = new Dictionary<string, string>();
+            environmentVariables.Add("PATH", path);
+            environmentVariables.Add("DOTNET_ROOT", dotnetRoot);
+            environmentVariables.Add("HELIX", helixQueue);
+
+            Console.WriteLine($"Current Directory: {HELIX_WORKITEM_ROOT}");
+            var helixDir = Environment.GetEnvironmentVariable("HELIX_WORKITEM_ROOT");
+            Console.WriteLine($"Setting HELIX_DIR: {helixDir}");
+            environmentVariables.Add("HELIX_DIR", helixDir);
+            environmentVariables.Add("NUGET_FALLBACK_PACKAGES", helixDir);
+            var nugetRestore = Path.Combine(helixDir, "nugetRestore");
+            Console.WriteLine($"Creating nuget restore directory: {nugetRestore}");
+            environmentVariables.Add("NUGET_RESTORE", nugetRestore);
+            var dotnetEFFullPath = Path.Combine(nugetRestore, $"dotnet-ef/{efVersion}/tools/netcoreapp3.1/any/dotnet-ef.exe");
+            Console.WriteLine($"Set DotNetEfFullPath: {dotnetEFFullPath}");
+            environmentVariables.Add("DotNetEfFullPath", dotnetEFFullPath);
+
+            Console.WriteLine("Checking for Microsoft.AspNetCore.App");
+            if (Directory.Exists("Microsoft.AspNetCore.App"))
+            {
+                Console.WriteLine($"Found Microsoft.AspNetCore.App, copying to {dotnetRoot}/shared/Microsoft.AspNetCore.App/{runtimeVersion}");
+                foreach (var file in Directory.EnumerateFiles("Microsoft.AspNetCore.App", "*.*", SearchOption.AllDirectories))
+                {
+                    File.Copy(file, $"{dotnetRoot}/shared/Microsoft.AspNetCore.App/{runtimeVersion}", overwrite: true);
+                }
+
+                Console.WriteLine($"Adding current directory to nuget sources: {HELIX_WORKITEM_ROOT}");
+
+                await ProcessUtil.RunAsync($"{dotnetRoot}/dotnet",
+                    $"nuget add source {HELIX_WORKITEM_ROOT} --configfile NuGet.config",
+                    environmentVariables: environmentVariables);
+
+                await ProcessUtil.RunAsync($"{dotnetRoot}/dotnet",
+                    "nuget add source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet5/nuget/v3/index.json --configfile NuGet.config",
+                    environmentVariables: environmentVariables);
+
+                await ProcessUtil.RunAsync($"{dotnetRoot}/dotnet",
+                    "nuget list source",
+                    environmentVariables: environmentVariables);
+
+                await ProcessUtil.RunAsync($"{dotnetRoot}/dotnet",
+                    $"tool install dotnet-ef --global --version {efVersion}",
```

Comment:
Global?

Reply:
We set the `DOTNET_CLI_HOME` environment variable so it isn't "global". But maybe I should move some more of that logic out of the scripts

Reply:
Yes